test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/BuildIntrospectionTest.cs

```diff
@@ -19,5 +19,37 @@ public class BuildIntrospectionTest : MSBuildIntegrationTestBase
             Assert.BuildOutputContainsLine(result, $"UpToDateCheckInput: {Path.Combine("Views", "Home", "Index.cshtml")}");
             Assert.BuildOutputContainsLine(result, $"UpToDateCheckInput: {Path.Combine("Views", "_ViewStart.cshtml")}");
         }
+
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task RazorSdk_SetsPreserveCompilationContextForProjectsWithCshtmlFiles()
+        {
+            var result = await DotnetMSBuild("Build", "/t:_IntrospectPreserveCompilationContext");
+
+            Assert.BuildPassed(result);
+            Assert.BuildOutputContainsLine(result, "PreserveCompilationContext: true");
+        }
+
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task RazorSdk_DoesNotSetPreserveCompilationContextForWebApplicationsWithoutCshtmlFiles()
+        {
+            Directory.Delete(Path.Combine(Project.DirectoryPath, "Views"), recursive: true);
+
+            var result = await DotnetMSBuild("Build", "/t:_IntrospectPreserveCompilationContext");
+
+            Assert.BuildPassed(result);
+            Assert.BuildOutputContainsLine(result, "PreserveCompilationContext: false");
+        }
+
+        [Fact]
+        [InitializeTestProject("ClassLibrary")]
+        public async Task RazorSdk_DoesNotSetPreserveCompilationContextForClassLibraryProjects()
```

Comment:
Writing this test requires creating a true ClassLibrary. The changes to support this get us in a better place and it's not overtly intrusive IMO

Reply:
Sure